src/java.base/share/classes/jdk/internal/classfile/components/CodeRelabeler.java

```diff
@@ -0,0 +1,147 @@
+ */
+package jdk.internal.classfile.components;
+
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+import jdk.internal.classfile.CodeBuilder;
+import jdk.internal.classfile.CodeElement;
+import jdk.internal.classfile.CodeTransform;
+import jdk.internal.classfile.instruction.BranchInstruction;
+import jdk.internal.classfile.instruction.LookupSwitchInstruction;
+import jdk.internal.classfile.instruction.SwitchCase;
+import jdk.internal.classfile.instruction.TableSwitchInstruction;
+import jdk.internal.classfile.Label;
+import jdk.internal.classfile.instruction.ExceptionCatch;
+import jdk.internal.classfile.instruction.LabelTarget;
+import jdk.internal.classfile.instruction.LocalVariable;
+import jdk.internal.classfile.instruction.LocalVariableType;
+
+/**
+ * CodeRelabeler is a {@link jdk.internal.classfile.CodeTransform} replacing all occurrences
+ * of {@link jdk.internal.classfile.Label} in the transformed code with new instances.
+ * All {@link jdk.internal.classfile.instruction.LabelTarget} instructions are adjusted accordingly.
+ * Relabeled code graph is identical to the original.
+ * <p>
+ * Primary purpose of CodeRelabeler is for repeated injections of the same code blocks.
+ * Repeated injection of the same code block must be relabeled, so each instance of
+ * {@link jdk.internal.classfile.Label} is bound in the target bytecode exactly once.
+ */
+public sealed interface CodeRelabeler extends CodeTransform {
+
+    /**
+     * Creates new instance of CodeRelabeler
+     * @return new instance of CodeRelabeler
+     */
+    static CodeRelabeler of() {
+        return of(new IdentityHashMap<>());
+    }
+
+    /**
+     * Creates new instance of CodeRelabeler storing the label mapping into the provided map
+     * @param map label map actively used for relabeling
+     * @return new instance of CodeRelabeler
+     */
+    static CodeRelabeler of(Map<Label, Label> map) {
+        return of((l, cob) -> map.computeIfAbsent(l, ll -> cob.newLabel()));
+    }
+
+    /**
+     * Creates new instance of CodeRelabeler using provided {@link java.util.function.BiFunction}
+     * to re-label the code.
+     * @param mapFunction
+     * @return
+     */
+    static CodeRelabeler of(BiFunction<Label, CodeBuilder, Label> mapFunction) {
+        return new CodeRelabelerImpl(mapFunction);
+    }
+
+    /**
+     * Access method to internal re-labeling function.
+     * @param label source label
+     * @param codeBuilder builder to create new labels
+     * @return target label
+     */
+    Label relabel(Label label, CodeBuilder codeBuilder);
+
+    record CodeRelabelerImpl(BiFunction<Label, CodeBuilder, Label> mapFunction) implements  CodeRelabeler {
+
+        @Override
+        public Label relabel(Label label, CodeBuilder cob) {
+            return mapFunction.apply(label, cob);
+        }
+
+        @Override
+        public void accept(CodeBuilder cob, CodeElement coe) {
+            switch (coe) {
```

Comment:
Missing case for`CharacterRange` instruction? I am not sure it makes any sense, if so perhaps add a comment as to why.

Reply:
Fixed, thanks.